sklearn/neighbors/kde.py

```diff
@@ -120,9 +131,61 @@ def fit(self, X, y=None):
         X : array_like, shape (n_samples, n_features)
             List of n_features-dimensional data points.  Each row
             corresponds to a single data point.
+
+        Warnings
+        --------
+        RuntimeWarning : A RuntimeWarning can be raised by 
+            sklearn.model_selection._search._store() This is caused by too small
+            initial bandwidth guasses of the grid search in cv. The grid search 
+            should still complete successfully. 
         """
         algorithm = self._choose_algorithm(self.algorithm, self.metric)
         X = check_array(X, order='C', dtype=DTYPE)
+        
+        if self.bandwidth_ == 'scott':
+            self.bandwidth =  X.shape[0]**(-1./(X.shape[1]+4)) 
+        elif self.bandwidth_ == 'silvermann':
+            self.bandwidth = (X.shape[0] * (X.shape[1] + 2) / 4.)**(-1. / (X.shape[1] + 4))
+        elif self.bandwidth_ == 'cv':
+            steps = 10
```

Comment:
I suspect this predetermines too many constants. But certainly it wastes computation rebuilding the tree in each underlying fit. It may also to be me conventional to provide KernelDensityCV to do this...